cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionConsumer.java

```diff
@@ -55,4 +56,13 @@
    * @param succeeded whether or not processing of the specified partitions was successful
    */
   void onFinish(List<? extends Partition> partitions, boolean succeeded);
+
+  /**
+   * Same as {@link #onFinish(List, boolean)}, but allows specifying {@link PartitionKey}s
+   * instead of {@link co.cask.cdap.api.dataset.lib.Partition}s.
+   *
+   * @param partitionKeys list of partition keys to mark as either succeeded or failed processing
+   * @param succeeded whether or not processing of the specified partitions was successful
+   */
+  void onFinishWithKeys(List<PartitionKey> partitionKeys, boolean succeeded);
```

Comment:
We need it to be List (instead of Iterable), because ordering matters.
I have made the change `PartitionKey` --> `? extends PartitionKey`, though.